feature/neighbor_set_inf(single feature)

TommasoRomani opened the pull request.

TommasoRomani committed:
aggiunto metodo di information theory

ncvescera commented on src/social_network_link_prediction/other_methods/neighbor_set_information.py:
```diff
@@ -0,0 +1,104 @@
+def prior(m, n, G):
+    kn = G.degree(n)
+    km = G.degree(m)
+    M = G.number_of_edges()
```
Lo richiami sempre e forse sprechi tempo.
Mettilo nel main e rendilo un parametro.

ncvescera replied:
Risolto in 08a9108

ncvescera commented on src/social_network_link_prediction/other_methods/neighbor_set_information.py:
```diff
@@ -0,0 +1,104 @@
+            }
+        s_xy.append(elem)
+    
+    return s_xy
```
Fai ritornare una matrice sparsa

ncvescera commented on src/social_network_link_prediction/other_methods/neighbor_set_information.py:
```diff
@@ -0,0 +1,104 @@
+    s_xy = []
+
+    for i,j in nx.complement(G).edges():
+        I_Oxy = overlap_info(G, i, j, prior, likelihood)
```
```suggestion
        I_Oxy = overlap_info(G, i, j)
```

ncvescera replied:
Risolto in 08a9108

ncvescera commented on src/social_network_link_prediction/other_methods/neighbor_set_information.py:
```diff
@@ -0,0 +1,104 @@
+# definition of the 2 indormation used in this information theory approach: overlapping nodes of different sets and
+# the exsistence of link across different sets 
+#
+def overlap_info(G:nx.Graph, x, y, prior, likelihood):
```
```suggestion
def overlap_info(G:nx.Graph, x, y):
```

ncvescera replied:
Risolto in 08a9108

TommasoRomani committed:
modifiche richieste apportate

ncvescera commented on src/social_network_link_prediction/other_methods/neighbor_set_information.py:
```diff
@@ -69,16 +68,14 @@ def likelihood(z, G):
 def MI(G : nx.Graph, lalla: int):
     I_Oxy = 0
     s_xy = []
+    res = np.zeros((G.number_of_nodes(), G.number_of_nodes()))
```
La matrice risultante sarà piena o può anche essere sparsa ??

Nel caso sarebbe da utilizzare direttamente una matrice sparsa.

ncvescera replied:
Risolto in 1b04464

ncvescera commented on src/social_network_link_prediction/other_methods/neighbor_set_information.py:
```diff
@@ -17,10 +17,10 @@
 # the exsistence of link across different sets 
 #
-def overlap_info(G:nx.Graph, x, y, prior, likelihood):
+def overlap_info(G:nx.Graph, x, y, edge_num):
```
nice 👍🏻

TommasoRomani committed:
Opera e ritorna correttamente una matrice sparsa

ncvescera commented on src/social_network_link_prediction/other_methods/neighbor_set_information.py:
```diff
@@ -0,0 +1,107 @@
+    for i,j in nx.complement(G).edges():
+        I_Oxy = overlap_info(G, i, j, edge_num)
+        res_sparse[i-1,j-1] = I_Oxy
+    
```
le operazioni `i-1` sono date dal fatto che i nomi dei nodi iniziano da 1?

TommasoRomani replied:
L'ho corretto con il metodo di networkx : nx.convert_node_labels_to_integers(), riga 91

TommasoRomani committed:
corretto il probelma degli indici

TommasoRomani committed:
fatte ultime modifiche

TommasoRomani committed:
corretto errore che faceva divergere soluzioni a -inf

ncvescera committed:
Apportati alcuni Fix

Prevenzione di ZeroDivitionError.
Creata direttamente la matrice sparsa senza passare da una FULL.
Aggiunti Type Hints.

ncvescera committed:
Usless Check Fix

Rimosso controllo inutil per ZeroDivitionError
(il ritardo mi ha colpito 🦧)

ncvescera reviewed (approved):
👍🏻

CristianCosci committed:
Update neighbor_set_information.py

Scritto TODO post merge
